File: T1/Navigation.kt
package com.example.p1

import androidx.compose.material.icons.Icons // Es bueno tenerlo para la estructura general
import androidx.compose.material.icons.automirrored.filled.List // Importación correcta para List auto-reflejado
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material.icons.filled.Edit
import androidx.compose.material.icons.filled.Info
import androidx.compose.material.icons.filled.PlayArrow
import androidx.compose.ui.graphics.vector.ImageVector

sealed class Screen(val route: String, val title: String, val icon: ImageVector) {
    object TextFields : Screen("textfields", "TextFields", Icons.Filled.Edit)
    object Buttons : Screen("buttons", "Botones", Icons.Filled.PlayArrow)
    object Selection : Screen("selection", "Selección", Icons.Filled.CheckCircle)
    object Lists : Screen("lists", "Listas", Icons.AutoMirrored.Filled.List) // Uso correcto
    object Info : Screen("info", "Info", Icons.Filled.Info)
}

File: Practica1/p1_flutter/navigation.dart
import 'package:flutter/material.dart';

// Una clase simple para definir la estructura de cada pantalla en la barra de navegación
class ScreenItem {
  final String title;
  final IconData icon;

  ScreenItem(this.title, this.icon);
}

// La lista de nuestras pantallas, equivalente a la sealed class 'Screen'
final List<ScreenItem> screenItems = [
  ScreenItem("TextFields", Icons.edit),
  ScreenItem("Botones", Icons.play_arrow),
  ScreenItem("Selección", Icons.check_circle),
  ScreenItem("Listas", Icons.list),
  ScreenItem("Info", Icons.info),
];